server/server.go

```diff
@@ -1305,7 +1305,12 @@ func (s *Server) fetchAccountClaims(name string) (*jwt.AccountClaims, string, er
 	if err != nil {
 		return nil, _EMPTY_, err
 	}
-	return s.verifyAccountClaims(claimJWT)
+	var claim *jwt.AccountClaims
+	claim, claimJWT, err = s.verifyAccountClaims(claimJWT)
+	if claim != nil && claim.Subject != name {
```

Comment:
Are you sure about that? (I think I recall once that you wanted to add some check and during a meeting you realized that assumption may have been wrong, but don't know if it was about that specifically).

Reply:
This extra check assures that we actually return what we asked for, not smth. we are made to believe we asked for.
Imagine this scenario. Requesting account key AA... but get a response AB..., without this check we would end up with an account we did not ask for.
Subject contains the account public key we are asking for.

Reply:
Ok